src/routing_table/mod.rs

```diff
@@ -396,10 +363,8 @@ impl<T: Binary + Clone + Copy + Debug + Default + Hash + Xorable> RoutingTable<T
         self.closest_names(name, count).is_some()
     }
 
-    /// Returns the `count` closest entries to `name` in the routing table, including our own name,
-    /// sorted by ascending distance to `name`. If we are not close, returns `None`.
-    pub fn closest_names(&self, name: &T, count: usize) -> Option<Vec<&T>> {
-        let result = self.groups
+    fn all_closest_names(&self, name: &T, count: usize) -> Vec<&T> {
```

Comment:
Maybe `closest_known_names`, as opposed to the real `closest_names`, where we know that they're the closest in the whole network?